### ISSUE TITLE
proposal: add support for DefaultFunc for string fields

### ISSUE THREAD

**Issue by haoxins** [facebook/ent#935]
Is it an accepted idea to add new build-in custom id type - cuid?

https://github.com/ericelliott/cuid

**Comment by EricMasonFa** [comment]
It should be possible:

```go
import "gopkg.in/lucsky/cuid.v1"

func (Pet) Fields() []ent.Field {
    return []ent.Field{
        field.String("id").
            MaxLen(25).
            NotEmpty().
            Unique().
            Immutable(),
            Default(cuid.New).
    }
}
```

**Comment by a8m** [comment]
Hey @haoxins,
I think @EricMasonFa suggestion sounds good. Is there anything specific you expect ent will provide for such case? If you have a custom Go type, you can use the [`GoType`](https://entgo.io/docs/schema-fields/#go-type) option.

**Comment by greyg00s** [comment]
@EricMasonFa I'm not sure if this works. 
```go 
field.String
```
 takes a default string and not a default function as the UUID type does. Unless I missed something

**Comment by a8m** [comment]
You're right @greyg00s, but we can add support for this. However, you should not be blocked on this, and can use [schema-hooks](https://entgo.io/docs/hooks/#schema-hooks) for setting default values on mutations.

**Comment by haoxins** [comment]
So, is it a good point to add a new func DefaultFunc (maybe another better name) to the FieldBuilder?

**Comment by a8m** [comment]
Yes, I think we can add it only for string fields for now and add it to others in the future if we'll see a need for this. 

Please let me know if you want to contribute it and need help with this 😃

**Comment by haoxins** [comment]
I'm pleasure to do that after my vacation :)

**Comment by a8m** [comment]
Closed with #1086, thanks @nolotz for the great work 😃
